### PR TITLE
Push progress bars - improve handling of push events

### PR DESCRIPTION
**Pull request by chekunkov**
For events without total progress take current as total.
For events with current progress greater than total - update total for the bar.

Fixes this:

push_empty_layer

### CODE REVIEW

File: tests/image/test_push.py [hunk 1]
```diff
@@ -52,25 +77,72 @@ def test_cli_with_progress(docker_client_mock, test_mock):
...
+        'Layers: 100%|██████████| 4/4\n\r'
+        'abc: 100%|██████████| 24.8K/24.8K [1.00MB/s]\n\n'
+        'The image registry/user/project:test pushed successfully.\n'
     )
-    matched = re.search(expected, result.output)
-    assert matched
-    # to make sure the test regexp below is correct
-    assert matched.group(0).startswith('Layers:  33%')
-    assert matched.group(0).endswith('pushed successfully.\n')
+    assert expected in result.output
```
Comment (vshlapakov) [inline tests/image/test_push.py]: Just curious: why not exact match? To skip potential warning messages at start?

File: tests/image/test_push.py [hunk 2]
```diff
@@ -15,6 +20,26 @@ def test_mock():
         yield m
 
 
+@pytest.fixture(autouse=True)
+def monkeypatch_bar_rate(monkeypatch):
```
Comment (vshlapakov) [inline tests/image/test_push.py]: All the monkeypatch logic is only to drop regexp below, right? I wasn't sure if it's worth it - as it brings more internals of tqdm, but It's less error-prone indeed and will be helpful to add more tests 👍

Comment (chekunkov) [inline tests/image/test_push.py]: When I started writing new test I realized that converting output to regex manually is too much, I want something simpler.

File: tests/image/test_push.py (hunk 1 above)
Comment (chekunkov) [inline tests/image/test_push.py]: Correct, I have HTTP warnings that broke the tests

**Pull request merged**